lib/index.js

```diff
@@ -321,3 +375,74 @@ function extendTraits(arr) {
 
   return arr;
 }
+
+/**
+ * Since Mixpanel doesn't support lists of objects, invert each list of objects to a set of lists of object properties.
+ * Treats list transformation atomically, e.g. will only transform if EVERY item in list is an object
+ *
+ * @api private
+ * @param {Object} props
+ * @example 
+ * input: {products: [{sku: 32, revenue: 99}, {sku:2, revenue: 103}]}
+ * output: {products_skus: [32, 2], products_revenues: [99, 103]}
+ */
+
+function invertObjLists(props) {
+  for (var propKey in props) {  // eslint-disable-line
+    var prop = props[propKey];
+    if (!props.hasOwnProperty(propKey) || !Array.isArray(prop)) {
+      continue;
+    }
+    var invertedLists = {};       
+    var listContainsNonObject = false;
+
+    // invert object lists and collect into invertedLists
+    for (var i=0; i<prop.length; i++) {
+      var elem = prop[i];
+      if (typeof elem !== 'object') {
+        listContainsNonObject = true;
+        break;
+      }
+      for (var key in elem) {
+        if (!elem.hasOwnProperty(key)) {
+          continue;
+        }
+        var attrKey = propKey+'_'+key+'s';  // e.g. products_skus
+        if (attrKey in invertedLists) {
```

Comment:
`invertedLists[attrKey]` ? invertedLists[attrKey].push() : invertedLists[attrKey] = [elem]

Reply:
decided to go with a comment here for readability, was .a lot to put in one line